src/instance/misc.js

```diff
@@ -39,6 +39,14 @@ exports._applyFilters = function (value, oldValue, filters, write) {
   return value
 }
 
+exports._bubbleChanges = function (value, expression) {
```

Comment:
It's more about responsibilities, IMO. Written this way, misc.js mixin is responsible for both forward and reverse filters, and abstracts forContext existence from the watcher.

Also, `alias === expression` does not look nice (because I easily bypassed this logic in my initial fiddle). Any ideas?